lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -323,6 +471,61 @@ struct ListCreateOpConversion
   }
 };
 
+struct PathOpConversion : public OpConversionPattern<firrtl::PathOp> {
+
+  PathOpConversion(TypeConverter &typeConverter, MLIRContext *context,
+                   const PathInfoTable &pathInfoTable,
+                   PatternBenefit benefit = 1)
+      : OpConversionPattern(typeConverter, context, benefit),
+        pathInfoTable(pathInfoTable) {}
+
+  LogicalResult
+  matchAndRewrite(firrtl::PathOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto *context = op->getContext();
+    auto pathType = om::PathType::get(context);
+    auto pathInfo = pathInfoTable.lookup(op.getTarget());
+
+    // If the target was optimized away, then replace the path operation with
+    // a deleted path.
+    if (!pathInfo) {
+      if (op.getTargetKind() == firrtl::TargetKind::DontTouch)
+        return emitError(op.getLoc(), "DontTouch target was deleted");
```

Comment:
Maybe `rewriter.notifyMatchFailure(...)`? (haven't tested, maybe that'll be combined/used with the error about legalization failing?)